### PR TITLE
Spark 245 - updated with SSSD support, updated AD and LDAP login support

### PR DESCRIPTION
**Pull request by madhusudhannr**
Spark245 is updated with following details,
- newer version of Livy server,
- Hadoop client.
- enhanced Jupyterhub with LDAP and AD log-in support
- sssd enabled.

### CODE REVIEW

File: deploy/example_clusters/cr-cluster-spark245.yaml [hunk 1]
```diff
@@ -5,15 +5,22 @@
...
 #  jupyterhub_lookup_dn: "True"
 #  jupyterhub_user_search_base: "dc=mip,dc=storage,dc=hpecorp,dc=net"
 #  jupyterhub_user_attribute: "cn"
 #  jupyterhub_bind_user_dn: "cn=admin,dc=mip,dc=storage,dc=hpecorp,dc=net"
 ##  #optional parameter. "use ":" as a delimiter to provide multiple templates"
 #  jupyterhub_bind_dn_template: "cn={username},ou=People,dc=mip,dc=storage,dc=hpecorp,dc=net"
-#  jupyterhub_valid_username_regex: "^[a-z][.@a-z0-9_-]*$"
+#  jupyterhub_valid_username_regex: "^[A-Za-z][.@A-Za-z0-9_-]*$"
+##  "use ":" as a delimiter to provide multiple jupyterhub_member_of"
+#  jupyterhub_member_of: CN=Eng,OU=Engineering,DC=mip,DC=storage,DC=hpecorp,DC=net:CN=QA,OU=Engineering,DC=mip,DC=storage,DC=hpecorp,DC=net
```
Comment (riteshja) [inline deploy/example_clusters/cr-cluster-spark245.yaml:23]: we should find some other way of getting this info. this is not user-friendly.

File: deploy/example_clusters/cr-cluster-spark245-stor.yaml [hunk 2]
```diff
@@ -13,7 +17,9 @@
 #  jupyterhub_bind_user_dn: "cn=admin,dc=mip,dc=storage,dc=hpecorp,dc=net"
 ##  #optional parameter. "use ":" as a delimiter to provide multiple templates"
 #  jupyterhub_bind_dn_template: "cn={username},ou=People,dc=mip,dc=storage,dc=hpecorp,dc=net"
-#  jupyterhub_valid_username_regex: "^[a-z][.@a-z0-9_-]*$"
+#  jupyterhub_valid_username_regex: "^[A-Za-z][.@A-Za-z0-9_-]*$"
+##  "use ":" as a delimiter to provide multiple jupyterhub_member_of"
+#  jupyterhub_member_of: CN=Eng,OU=Engineering,DC=mip,DC=storage,DC=hpecorp,DC=net:CN=QA,OU=Engineering,DC=mip,DC=storage,DC=hpecorp,DC=net
```
Comment (riteshja) [inline deploy/example_clusters/cr-cluster-spark245-stor.yaml:22]: same comment as below

File: deploy/example_clusters/cr-cluster-spark245.yaml (hunk 1 above)
Comment (kmathur2) [reply deploy/example_clusters/cr-cluster-spark245.yaml:23]: are we using connection for this?

File: deploy/example_clusters/cr-cluster-spark245-stor.yaml (hunk 2 above)
Comment (kmathur2) [reply deploy/example_clusters/cr-cluster-spark245-stor.yaml:22]: we should not put our AD/LDAP info in our apps

File: deploy/example_clusters/cr-cluster-spark245.yaml (hunk 1 above)
Comment (madhusudhannr) [reply deploy/example_clusters/cr-cluster-spark245.yaml:23]: @riteshja  
As discussed, ConfigMap accepts only String. arry gives error.

@kmathur2 
Once configcli with connection feature is available as part of Kubedirector EPIC, code will be updated to use configcli with connections.
need to update kubedirector to use new version of configcli and EPIC to have updated KD (which is having new configcli)

File: deploy/example_clusters/cr-cluster-spark245-stor.yaml (hunk 2 above)
Comment (madhusudhannr) [reply deploy/example_clusters/cr-cluster-spark245-stor.yaml:22]: done. changed to some dummy values

**Review by riteshja — APPROVED** [review]

File: deploy/example_clusters/cr-cluster-spark245-stor.yaml [hunk 3]
```diff
@@ -1,29 +1,36 @@
...
+#  user_search_base: "ou=people,dc=wikimedia,dc=org"
+#  user_attribute: "cn"
+#  bind_user_dn: "cn=admin,ou=people,dc=wikimedia,dc=org"
 ##  #optional parameter. "use ":" as a delimiter to provide multiple templates"
-#  jupyterhub_bind_dn_template: "cn={username},ou=People,dc=mip,dc=storage,dc=hpecorp,dc=net"
-#  jupyterhub_valid_username_regex: "^[a-z][.@a-z0-9_-]*$"
+#  bind_dn_template: "cn={username},ou=People,ou=people,dc=wikimedia,dc=org"
+#  valid_username_regex: "^[A-Za-z][.@A-Za-z0-9_-]*$"
+##  "use ":" as a delimiter to provide multiple member_of"
+#  member_of: CN=Eng,OU=Engineering,ou=people,dc=wikimedia,dc=org:CN=QA,OU=Engineering,ou=people,dc=wikimedia,dc=org
```
Comment (riteshja) [inline deploy/example_clusters/cr-cluster-spark245-stor.yaml:23]: this is still old format

File: deploy/example_clusters/cr-cluster-spark245.yaml [hunk 4]
```diff
@@ -1,29 +1,38 @@
...
+#  user_attribute: "cn"
+#  bind_user_dn: "cn=admin,ou=people,dc=wikimedia,dc=org"
 ##  #optional parameter. "use ":" as a delimiter to provide multiple templates"
-#  jupyterhub_bind_dn_template: "cn={username},ou=People,dc=mip,dc=storage,dc=hpecorp,dc=net"
-#  jupyterhub_valid_username_regex: "^[a-z][.@a-z0-9_-]*$"
+#  bind_dn_template: "cn={username},ou=People,ou=people,dc=wikimedia,dc=org"
+#  valid_username_regex: "^[A-Za-z][.@A-Za-z0-9_-]*$"
+##  "use ":" as a delimiter to provide multiple member_of"
+#  member_of:
+#     - CN=Eng,OU=Engineering,ou=people,dc=wikimedia,dc=org
```
Comment (riteshja) [inline deploy/example_clusters/cr-cluster-spark245.yaml:24]: update to this format in the above yaml file

Comment (madhusudhannr) [reply deploy/example_clusters/cr-cluster-spark245.yaml]: corrected.

File: deploy/example_clusters/cr-cluster-spark245-stor.yaml (hunk 3 above)
Comment (madhusudhannr) [reply deploy/example_clusters/cr-cluster-spark245-stor.yaml:23]: done.

**Comment by kmathur2** [comment]
Is this good to go? Please resolve conflicts

**Comment by madhusudhannr** [comment]
> Is this good to go? Please resolve conflicts

resolved the conflict.

**Comment by riteshja** [comment]
This is good to go

**Pull request closed without merging**